src/main.c

```diff
@@ -1,50 +1,145 @@
 #include <clang-c/Index.h>
 #include <stdio.h>
 
-void printCXString(CXString s)
+enum CXChildVisitResult visitor(CXCursor, CXCursor, CXClientData); 
+enum CXChildVisitResult enum_child_visitor(CXCursor, CXCursor, CXClientData); 
+enum CXChildVisitResult struct_child_visitor(CXCursor, CXCursor, CXClientData); 
+void enum_handler(CXCursor);
```

Comment:
These two handlers should receive all three parameters so you can avoid calling clang_getCursorLexicalParent. The third parameter will be necessary when we introduce Lua.

Reply:
Ok, will add that.

Reply:
I used `clang_getCursorLexicalParent()` as `parent` behaved ambiguously in some case which I don't remember. But now, it is doing fine. Will find out while testing.